projects/pipelines/visualizations/rhods-pipelines/store/lts_parser.py

```diff
@@ -134,26 +139,48 @@ def _generate_run_duration(results):
                         data[resource_key] = []
                     data[resource_key].append((user_data.complete_run_times[workflow_run_name] - user_data.workflow_start_times[resource_name.split("/")[1]]).total_seconds())
 
-    run_duration = _generate_dsp_test_stats(list(reduce(lambda l, r: l + r, data.values(), [])))
-    medians = np.array([np.median(d[1]) for d in sorted(data.items())])
-    A = np.vstack([np.arange(len(medians)), np.ones(len(medians))]).T
-    slope, _ = np.linalg.lstsq(A, medians, rcond=None)[0]
-    run_duration["degrade_speed"] = slope
+    all_points = list(reduce(lambda l, r: l + r, data.values(), []))
+    run_duration = _generate_dsp_test_stats(all_points)
+    if len(all_points) <= 0:
+        run_duration["degrade_speed"] = 0.0
+    else:
+        medians = np.array([np.median(d[1]) for d in sorted(data.items())])
+        A = np.vstack([np.arange(len(medians)), np.ones(len(medians))]).T
+        slope, _ = np.linalg.lstsq(A, medians, rcond=None)[0]
+        run_duration["degrade_speed"] = slope
 
     return types.SimpleNamespace(**run_duration)
 
-def _generate_dsp_test_stats(data: List[float]):
+def _generate_dsp_test_stats(data: List[float]) -> Dict[str, float]:
     test_stats = {}
 
-    test_stats["values"] = data
-    test_stats["min"] = np.min(data)
-    test_stats["max"] = np.max(data)
-    test_stats["median"] = np.median(data)
-    test_stats["mean"] = np.mean(data)
-    test_stats["percentile_80"] = np.percentile(data, 80)
-    test_stats["percentile_90"] = np.percentile(data, 90)
-    test_stats["percentile_95"] = np.percentile(data, 95)
-    test_stats["percentile_99"] = np.percentile(data, 99)
+    # Some of the Numpy functions panic on 0-length arrays
+    # so we need to check this explicitly
+    if len(data) <= 0:
```

Comment:
```suggestion
    if not data:
```

Reply:
Personally I don't like relying on the implicit bool conversion in Python. Happy to make the switch in the name of consistency though.

Reply:
yes, please update if you don't mind